packages/rspack/src/builtin-plugin/css-extract/index.ts

```diff
@@ -0,0 +1,124 @@
+import type { RawCssExtractPluginOption } from "@rspack/binding";
+import { Compiler } from "../..";
+import { MODULE_TYPE } from "./loader";
+
+export * from "./loader";
+
+const DEFAULT_FILENAME = "[name].css";
+const LOADER_PATH = require.resolve("./loader");
+
+export interface PluginOptions {
+	filename?: string;
+	chunkFilename?: string;
+	ignoreOrder?: boolean;
+	insert?: string | ((linkTag: HTMLLinkElement) => void);
+	attributes?: Record<string, string>;
+	linkType?: string | "text/css" | false;
+	runtime?: boolean;
+
+	// workaround for pathinto, deprecate this when rspack supports pathinfo
+	pathinfo?: boolean;
+}
+
+export class CssExtractRspackPlugin {
+	static pluginName: string = "css-extract-rspack-plugin";
+	static loader: string = LOADER_PATH;
+
+	options: PluginOptions;
+
+	constructor(options?: PluginOptions) {
```

Comment:
if this is not working with newTreeshaking:false, we should add warning when newTreeshaking is set to true @JSerFeng